src/lib/sql/sql_translator.cpp

```diff
@@ -1051,6 +1054,80 @@ SQLTranslator::TableSourceState SQLTranslator::_translate_predicated_join(const
   return result_state;
 }
 
+SQLTranslator::TableSourceState SQLTranslator::_translate_named_columns_join(const hsql::JoinDefinition& join) {
+  const auto join_mode = translate_join_mode(join.type);
+
+  auto left_state = _translate_table_ref(*join.left);
+  const auto& right_state = _translate_table_ref(*join.right);
+
+  const auto left_input_lqp = left_state.lqp;
+  const auto& right_input_lqp = right_state.lqp;
+
+  Assert(!join.namedColumns->empty(), "Expected at least one named column.");
+  Assert(!join.condition, "Did not expect join condition for join using named columns.");
+
+  const auto& left_sql_identifier_resolver = left_state.sql_identifier_resolver;
+  const auto& right_sql_identifier_resolver = right_state.sql_identifier_resolver;
+
+  Assert(left_sql_identifier_resolver, "Expected SQLIdentifierResolver for left input.");
+  Assert(right_sql_identifier_resolver, "Expected SQLIdentifierResolver for right input.");
+
+  auto join_predicates = std::vector<std::shared_ptr<AbstractExpression>>{join.namedColumns->size()};
```

Comment:
Note for me: We have to write that down in the coding guidelines.

You're absolutely fine, we prefer curly braces for initialization. *BUT* some STL structures (such as `std::vector`) differ between the semantics: `std::vector<int>(2)` creates a vector of two default-initialized ints (`0`s), `std::vector<int>{2}` creates a vector with a single `2`. Thus, we use regular braces when it comes to the number of items to make that clear. Sure, it's no difference for a vector of pointers, but things tend to slip through.